platform/contracts/admin/src/contracts/mod.rs

```diff
@@ -21,22 +21,58 @@ mod impl_mod;
 mod platform;
 mod protocol;
 
-#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, JsonSchema)]
-#[serde(rename_all = "snake_case", deny_unknown_fields)]
-pub struct ContractsTemplate<T, U = ProtocolTemplate<BTreeMap<String, T>>> {
-    pub platform: PlatformTemplate<T>,
-    pub protocol: U,
+pub trait Hkt {
```

Comment:
what does Hkt stand for?

Reply:
Higher-kind type. Another name for it is higher-order type. I can rename it to not use the abbreviation, if you think it'd be easier to understand.

Reply:
this abbreviature is not so widely known, at least from my point of view, what short-and-descriptive names would you come with?

Reply:
`TypeConstructor` is one I can think of as a replacement.